copy/copy.go

```diff
@@ -1195,10 +1203,9 @@ func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, to
 				Cache:         ic.c.blobInfoCache,
 				CanSubstitute: ic.canSubstituteBlobs,
 				SrcRef:        srcRef,
+				EmptyLayer:    emptyLayer,
 			}
-			if enableEarlyCommit {
-				options.LayerIndex = &layerIndex
-			}
+			options.LayerIndex = &layerIndex
```

Comment:
(Absolutely non-blocking: maybe fold this into the initializer above — OTOH this way it’s more consistent with the other use.)

BTW random unrelated notes to self for that some-time-maybe issue:
- `srcRef` does not need to be a parameter passed down the stack; it could be computed once when creating `copier` instead of per every layer
- It _almost_ seems that `copyLayer` and callees could pass around a `PutBlobOptions` instead of a long list of unnamed parameters. Making that would would probably be a bit of churn (the -1/nil difference for `emptyLayer`, probably consolidate `PutBlobOptions` and `TryReusingBlobOptions`) but might pay off in readability.